proposals/0390-noncopyable-structs-and-enums.md

```diff
@@ -245,6 +245,9 @@ protocol Foo where Self: ~Copyable {
   // ERROR: associated type requirements must be `Copyable`
   associatedtype Bar: ~Copyable
 }
+
+// ERROR: cannot suppress `Copyable` in extension of `FileWithPath`
+extension FileWithPath: ~Copyable {}
```

Comment:
hm. good point. While that syntax wouldn't currently be valid for `FileWithPath`, I'll leave this point about not allowing it in extensions "currently" as-is, since we're not sure how we want people to express that idea. Personally I think it should be fine for people to just say `extension Optional { ... }` and all methods in there apply to all Optionals, whether they're copyable or not. Thus, methods in there would get checks and errors about copying a noncopyable `self`, etc, since that's the most general kind of Optional.

Reply:
Thus, I was thinking you have to explicitly exclude the most general kind of noncopyable optional via `extension Optional: Copyable { ... }` or something like that.